crates/holochain/src/core/workflow/app_validation_workflow/validation_dependencies.rs

```diff
@@ -0,0 +1,289 @@
+use std::{
+    collections::{HashMap, HashSet},
+    time::{Duration, Instant},
+};
+
+use holo_hash::{AnyDhtHash, DhtOpHash};
+use holochain_types::dht_op::DhtOpHashed;
+
+/// In-memory struct to keep track of missing DHT hashes, which DhtOp depends on them
+/// and when they were started being fetched.
+pub struct ValidationDependencies {
+    /// Missing hashes that are being fetched, along with a set of DhtOps that depend
+    /// on the hash and the last Instant a fetch was attempted.
+    missing_hashes: HashMap<AnyDhtHash, (HashSet<DhtOpHash>, Instant)>,
```

Comment:
`missing_hashes` was hash map of missing hashes with the instant they were first fetched
`hashes_missing_for_op` was a hash map of op hash with the corresponding dht hashes to fetch

=> combined into `missing_hashes`, a hash map of the missing dht hash with the ops that depend on it and the instant it was first fetched

Reply:
The structure looks good but it is a more complex type. I'd be tempted to introduce a type to wrap the `(HashSet<DhtOpHash>, Instant)` but let's not mess with it as part of this PR :)

Reply:
I was tempted to do that and didn't in the end. I'll do it with your encouragement. =)